src/OpenTK/Platform/MacOS/Cocoa/NSAutoreleasePool.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+namespace OpenTK.Platform.MacOS
+{
+    /// <summary>
+    /// The <see cref="NSAutoreleasePool"/> class is a wrapper around the native objective-C NSAutoreleasePool.
+    /// In particular, this construct mimics the usage of an @autorelease block and can be used in much the same way,
+    /// only with a C# using block instead.
+    /// </summary>
+    public sealed class NSAutoreleasePool : IDisposable
+    {
+        private readonly IntPtr _autoreleasePool;
+
+        /// <summary>
+        /// Allocates and initializes a new <see cref="NSAutoreleasePool"/>.
+        /// </summary>
+        public NSAutoreleasePool()
+        {
+            _autoreleasePool = Cocoa.SendIntPtr(Class.NSAutoreleasePool, Selector.Get("alloc"));
+            _autoreleasePool = Cocoa.SendIntPtr(_autoreleasePool, Selector.Get("init"));
```

Comment:
Repeat assignment is a little odd here. Separate var for the first alloc call?
Also you can reuse init and alloc selectors from Selector.